### PR TITLE
Add table aws_emr_block_public_access_configuration

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
# Integration test logs
<details>
No integration test exist.
</details>

# Example query results
<details>

```pgsql
select
  created_by_arn,
  block_public_security_group_rules,
  creation_date,
  classification,
  configurations,
  permitted_public_security_group_rule_ranges
from
  aws_emr_block_public_access_configuration
order by
  created_by_arn,
  creation_date;
```
Result json:
```
[
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "arn:aws:iam::222222222222:user/madhushree",
  "creation_date": "2023-02-21T18:55:54+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "block_public_security_group_rules": true,
  "classification": null,
  "configurations": null,
  "created_by_arn": "arn:aws:sts::222222222222:federated-user/subhajit",
  "creation_date": "2021-07-16T19:29:33+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 }
]
> 

```

```pgsql
select
  created_by_arn,
  creation_date,
  configurations,
  permitted_public_security_group_rule_ranges
from
  aws_emr_block_public_access_configuration
where 
  block_public_security_group_rules;
```
Result in json

```
[
 {
  "configurations": null,
  "created_by_arn": "arn:aws:iam::222222222222:user/madhushree",
  "creation_date": "2023-02-21T18:55:54+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "arn:aws:sts::222222222222:federated-user/subhajit",
  "creation_date": "2021-07-16T19:29:33+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 },
 {
  "configurations": null,
  "created_by_arn": "",
  "creation_date": "1970-01-01T05:30:00+05:30",
  "permitted_public_security_group_rule_ranges": [
   {
    "MaxRange": 22,
    "MinRange": 22
   }
  ]
 }
]
```
```
select
  created_by_arn,
  creation_date,
  rules ->> 'MaxRange' as max_range,
  rules ->> 'MinRange' as min_range
from
  aws_emr_block_public_access_configuration
  cross join jsonb_array_elements(permitted_public_security_group_rule_ranges) as rules;
+---------------------------------------------------+---------------------------+-----------+-----------+
| created_by_arn                                    | creation_date             | max_range | min_range |
+---------------------------------------------------+---------------------------+-----------+-----------+
| arn:aws:iam::222222222222:user/madhushree         | 2023-02-21T18:55:54+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
| arn:aws:sts::222222222222:federated-user/subhajit | 2021-07-16T19:29:33+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
|                                                   | 1970-01-01T05:30:00+05:30 | 22        | 22        |
+---------------------------------------------------+---------------------------+-----------+-----------+
```
```
select
  created_by_arn,
  creation_date,
  configurations
from
  aws_emr_block_public_access_configuration
where
  date_part('day', now() - creation_date) < 90
+-------------------------------------------+---------------------------+----------------+
| created_by_arn                            | creation_date             | configurations |
+-------------------------------------------+---------------------------+----------------+
| arn:aws:iam::222222222222:user/madhushree | 2023-02-21T18:55:54+05:30 | <null>         |
+-------------------------------------------+---------------------------+----------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_emr_block_public_access_configuration.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsEmrBlockPublicAccessConfiguration(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_emr_block_public_access_configuration",
+		Description: "AWS EMR Block Public Access Configuration",
+		List: &plugin.ListConfig{
```
Comment (ParthaI) [inline aws/table_aws_emr_block_public_access_configuration.go:21]: We don't have any get call for it?

**Review by ParthaI — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please manage the table column ordering.
- We don't have any standard column for it?

File: aws/table_aws_emr_block_public_access_configuration.go (hunk 1 above)
Comment (Priyanka-Chatterjee-2000) [reply aws/table_aws_emr_block_public_access_configuration.go:21]: No

**Review by ParthaI — APPROVED** [review]

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@Priyanka-Chatterjee-2000 Please see review comments, thanks!

File: docs/tables/aws_emr_block_public_access_configuration.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  classification,
+  configurations,
+  permitted_public_security_group_rule_ranges
+from
+  aws_emr_block_public_access_configuration
+order by
+  created_by_arn,
+  creation_date;
+```
+### List clusters with Block Public Access enabled
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply docs/tables/aws_emr_block_public_access_configuration.md:23]: Right it is not set on clusters individually so changing the description to `List Block Public Access enabled settings for EMR` is this good?

File: docs/tables/aws_emr_block_public_access_configuration.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  created_by_arn,
+  creation_date,
+  configurations
+from
+  aws_emr_block_public_access_configuration
+where 
+  block_public_security_group_rules;
+```
+
+### List Block Public Access with Permitted Public Security Group Rule Max and Min Ranges
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply docs/tables/aws_emr_block_public_access_configuration.md:36]: In Permitted Public Security Group Rule json array  we get max and min port ranges. Port number can be set anything based  on the way the user wish to access the instance.Here port 22, is used for SSH access to the cluster EC2 instances. If the inbound traffic security rules are set with IPv4 0.0.0.0/0 from port 22 then these ports are publicly accessible and any body will be able to connect to cluster EC2 instances through ssh but here we only get the data of port numbers and nothing else. As there are less number of columns and I was asked to add more example queries I added this. Shall I remove this?

File: docs/tables/aws_emr_block_public_access_configuration.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Table: aws_emr_block_public_access_configuration
+
+The `aws_emr_block_public_access_configuration` table provides Amazon EMR block public access configuration for your AWS account in the current Region.
+
+## Examples
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply docs/tables/aws_emr_block_public_access_configuration.md:5]: Added an example query with `configuration details`

**Review by misraved — APPROVED** [review]

**Pull request merged**